lib/smack/SmackInstGenerator.cpp

```diff
@@ -646,10 +646,8 @@ void SmackInstGenerator::visitCallInst(llvm::CallInst &ci) {
   } else if (name.find(Naming::RUST_ENTRY) != std::string::npos) {
     // Set the entry point for Rust programs
     auto castExpr = ci.getArgOperand(0);
-    if (auto CE = dyn_cast<const Constant>(castExpr)) {
-      auto mainFunc = CE->getOperand(0);
-      emit(Stmt::call(mainFunc->getName(), {}, {}));
-    }
+    auto mainFunction = dyn_cast<const Function>(castExpr);
```

Comment:
Why not use `cast<>` if we don't check for possible nullptr?

Reply:
Yes, this could be changed to `cast`.

Reply:
Done.